### PR TITLE
Remove ITransaction interface and update Transaction to follow specs

### PR DESCRIPTION
**Pull request by danielailie**

### CODE REVIEW

File: src/transaction.local.net.spec.ts [hunk 1]
```diff
@@ -40,16 +39,16 @@ describe("test transaction", function () {
         let transactionOne = new Transaction({
             sender: alice.address,
             receiver: bob.address,
-            value: TokenTransfer.egldFromAmount(42),
-            gasLimit: network.MinGasLimit,
+            value: TokenTransfer.egldFromAmount(42).amount,
```
Comment (popenta) [inline src/transaction.local.net.spec.ts:42]: let's not use this method

Comment (danielailie) [reply src/transaction.local.net.spec.ts:42]: I'll update in a different pr if that's ok

File: src/proto/serializer.spec.ts [hunk 2]
```diff
@@ -59,12 +57,12 @@ describe("serialize transactions", () => {
 
     it("with data, with value", async () => {
         let transaction = new Transaction({
-            nonce: 91,
-            value: TokenTransfer.egldFromAmount(10),
+            nonce: 91n,
+            value: TokenTransfer.egldFromAmount(10).amount,
```
[the remark this answers is not in the document]
Comment (danielailie) [reply src/proto/serializer.spec.ts:61]: I'll update in a different pr if that's ok

File: src/abi/smartContract.ts [hunk 3]
```diff
@@ -289,7 +289,7 @@ export class SmartContract implements ISmartContract {
      * @param nonce The owner nonce used for the deployment transaction
      */
     static computeAddress(owner: Address, nonce: INonce): Address {
-        const deployer = Address.fromBech32(owner.bech32());
+        const deployer = Address.fromBech32(owner.toBech32());
```
Comment (popenta) [inline src/abi/smartContract.ts:292]: here it seems that we create an Address from an Address.

**Review by popenta — APPROVED** [review]

File: src/abi/nativeSerializer.ts [hunk 4]
```diff
@@ -403,9 +403,9 @@ export namespace NativeSerializer {
     export function convertNativeToAddress(
         native: NativeTypes.NativeAddress,
         errorContext: ArgumentErrorContext,
-    ): IAddress {
+    ): Address {
         if ((<any>native).bech32) {
-            return <IAddress>native;
+            return <Address>native;
```
[the remark this answers is not in the document]
Comment (danielailie) [reply src/abi/nativeSerializer.ts:408]: after internal discussion it is ok as it is

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**